lms/djangoapps/course_api/blocks/api.py

```diff
@@ -75,6 +76,10 @@ def get_blocks(
     include_gated_sections = 'show_gated_sections' in requested_fields
     include_has_scheduled_content = 'has_scheduled_content' in requested_fields
     include_special_exams = 'special_exam_info' in requested_fields
+    include_discussions_context = (
+        DiscussionsTopicLinkTransformer.EMBED_URL in requested_fields or
+        DiscussionsTopicLinkTransformer.EXTERNAL_ID in requested_fields
```

Comment:
`DiscussionsTopicLinkTransformer.EMBED_URL` and `DiscussionsTopicLinkTransformer.EXTERNAL_ID` are passed from the frontend?

Reply:
Yes, we need to explicitly request that this field be included for the response to be returned. This is true of most fields that this API returns, which aren't returned by default.